src/Orleans.Core/Serialization/SerializationManager.cs

```diff
@@ -222,6 +222,22 @@ public void Register(Type t, DeepCopier cop, Serializer ser, Deserializer deser)
             Register(t, cop, ser, deser, false);
         }
 
+        private object InitializeSerializer(Type serializerType)
+        {
+            if (this.ServiceProvider == null)
+                return null;
```

Comment:
Hmmm, this shortcut can be error prone, as there are cases where it is valid to return `null` at runtime...

Reply:
I could throw in the GetSerializer/Copier/Deserializer methods if the `ServiceProvider` wasn't set?

Reply:
It's fine, since we should be removing this statement entirely in https://github.com/dotnet/orleans/pull/3669 now that we'll stop using the SM for the codegen path. /cc @ReubenBond

Reply:
BTW, @ReubenBond, remember to remove this and not allow a null service provider once you rebase your other PR to not use `SerializationManager` for codegen.